shared/src/main/java/uk/gov/di/authentication/shared/services/DynamoIdentityService.java

```diff
@@ -11,42 +11,44 @@
 import java.util.Objects;
 import java.util.Optional;
 
-import static uk.gov.di.authentication.shared.dynamodb.DynamoClientHelper.tableConfig;
+import static uk.gov.di.authentication.shared.dynamodb.DynamoClientHelper.createDynamoEnhancedClient;
 
 public class DynamoIdentityService {
 
     private static final String IDENTITY_CREDENTIALS_TABLE = "identity-credentials";
-    private final DynamoDBMapper identityCredentialsMapper;
     private final long timeToExist;
-    private final AmazonDynamoDB dynamoDB;
+    private final DynamoDbTable<IdentityCredentials> dynamoIdentityCredentialsTable;
 
     public DynamoIdentityService(ConfigurationService configurationService) {
         var tableName = configurationService.getEnvironment() + "-" + IDENTITY_CREDENTIALS_TABLE;
 
         this.timeToExist = configurationService.getAccessTokenExpiry();
-        this.dynamoDB = DynamoClientHelper.createDynamoClient(configurationService);
-        this.identityCredentialsMapper = new DynamoDBMapper(dynamoDB, tableConfig(tableName));
+        var dynamoDbEnhancedClient = createDynamoEnhancedClient(configurationService);
+        dynamoIdentityCredentialsTable =
+                dynamoDbEnhancedClient.table(
+                        tableName, TableSchema.fromBean(IdentityCredentials.class));
 
-        warmUp(tableName);
+        warmUp();
     }
 
     public void addCoreIdentityJWT(String subjectID, String coreIdentityJWT) {
         var identityCredentials =
-                identityCredentialsMapper.load(IdentityCredentials.class, subjectID);
+                dynamoIdentityCredentialsTable.getItem(
+                        Key.builder().partitionValue(subjectID).build());
         if (Objects.isNull(identityCredentials)) {
```

Comment:
I don't believe you need to check if it already exists in dynamo, `updateItem(...)` should create a new item if it doesn't already exist https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/enhanced/dynamodb/DynamoDbTable.html#updateItem-T-